src/inmanta/util/dict_path.py

```diff
@@ -223,6 +224,17 @@ def get_elements(self, container: object) -> list[object]:
         :param container: the container to search in
         """
 
+    @abc.abstractmethod
+    def get_paths(self, container: object) -> Sequence["DictPath"]:
+        """
+        Get all the dict paths that this wild dict path contains, in the given
+        container.  That is, the set of dict path that would resolve the same
+        set of elements as this wild dict path, when searching in the given
+        container.
```

Comment:
`when searching in the given container.` is a very weak constraint. 

i.e. when the container has 0 or 1 element, an empty filter is always correct. 

The intended usage is can't be supported under this definition.

I think you have to assume the path describes an identity and that identity is made concrete based on the instances present.

Reply:
Another way of seeing it is: the context I provided above is there to explain why I want this feature but the feature itself is not aware of this context, or lsm, and any notion of identity. Dict path makes it easy to interact with json-like structure with an identity mechanism defined, but doesn't enforce anything like that, so I don't think we should start with it now